tests/composition/pec/test_parameterestimationcomposition.py

```diff
@@ -228,14 +273,171 @@ def reward_rate(sim_data):
+
+    if func_mode == "Python":
+        pytest.skip(
+            "Test not yet implemented for Python. Parameter estimate is too slow."
+        )
+
+    def _run_ddm_with_params(
+            starting_value,
+            rate,
+            noise,
+            threshold,
+            non_decision_time,
+            time_step_size,
+            trial_inputs,
+    ):
+        """Create a composition with DDM and run it with the given parameters."""
+
+        # Create a simple one mechanism composition containing a DDM in integrator mode.
+        decision = pnl.DDM(
+            function=pnl.DriftDiffusionIntegrator(
+                starting_value=starting_value,
+                rate=rate,
+                noise=noise,
+                threshold=threshold,
+                non_decision_time=non_decision_time,
+                time_step_size=time_step_size,
+            ),
+            output_ports=[pnl.DECISION_OUTCOME, pnl.RESPONSE_TIME],
+            name="DDM",
+        )
+
+        comp = pnl.Composition(pathways=decision)
+
+        # Run the composition to generate some data to fit
+        comp.run(inputs={decision: trial_inputs})
+        results = comp.results
+
+        data_to_fit = pd.DataFrame(
+            np.squeeze(np.array(results)), columns=["decision", "response_time"]
+        )
+        data_to_fit["decision"] = data_to_fit["decision"].astype("category")
+
+        return comp, data_to_fit
+
+    # High-level parameters the impact performance of the test
+    num_trials = 50
+    time_step_size = 0.01
+    num_estimates = 20
+
+    # Let's generate an "experimental" dataset to fit. This is a parameter recovery test
+    # Lets make 10% of the trials have a positive stimulus drift rate, and the other 90%
+    # have a negative stimulus drift rate.
+    rng = np.random.default_rng(12345)
+    trial_inputs = rng.choice(
+        [5.0, -5.0], size=(num_trials, 1), p=[0.10, 0.9], replace=True
+    )
+
+    # Make the first and last input positive for sure. This helps make sure inputs are really getting
+    # passed to the composition correctly during parameter fitting, and we aren't just getting a single
+    # trials worth of a cached input.
+    trial_inputs[0] = np.abs(trial_inputs[0])
+    trial_inputs[-1] = np.abs(trial_inputs[-1])
+
+    ddm_params = dict(
+        starting_value=0.0,
+        rate=0.3,
+        noise=1.0,
+        threshold=0.6,
+        non_decision_time=0.15,
+        time_step_size=time_step_size,
+    )
+
+    # We will generate a dataset that comprises two different conditions. Each condition will have a different
+    # threshold.
+    params_cond1 = dict(
+        threshold=0.7,
+    )
+
+    params_cond2 = dict(
+        threshold=0.3,
+    )
+
+    comp, data_cond1 = _run_ddm_with_params(**{**ddm_params, **params_cond1}, trial_inputs=trial_inputs)
+    _, data_cond2 = _run_ddm_with_params(**{**ddm_params, **params_cond2}, trial_inputs=trial_inputs)
```

Comment:
The intermediate dict should not be necessary.
```
_run_ddm_with_params(**ddm_param, **params_cond2, ...)
```

Reply:
Fixed.